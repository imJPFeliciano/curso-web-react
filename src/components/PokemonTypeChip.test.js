import React from 'react';
import TypeChip from './PokemonTypeChip';
import { mount } from 'enzyme';
import 'jest-styled-components';

describe('PokemonTypeChip', () => {
  it('render sin problemas', () => {
    const wrapper = mount(<TypeChip name="grass" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('se muestra el color morado para tipo veneno', () => {
    const wrapper = mount(<TypeChip name="poison" />);
    expect(wrapper).toHaveStyleRule('background-color', 'purple');
  });

  it('se muestra el color verde para tipo hierba', () => {
    const wrapper = mount(<TypeChip name="grass" />);
    expect(wrapper).toHaveStyleRule('background-color', 'green');
  });

  it('se muestra el color amarillo para tipo electrico', () => {
    const wrapper = mount(<TypeChip name="electric" />);
    expect(wrapper).toHaveStyleRule('background-color', 'yellow');
  });
});