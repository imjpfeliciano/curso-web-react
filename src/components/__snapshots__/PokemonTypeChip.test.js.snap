// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonTypeChip render sin problemas 1`] = `ReactWrapper {}`;
